sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDeviceMemoryStore.scala

```diff
@@ -67,22 +67,18 @@ class RapidsDeviceMemoryStore(catalog: RapidsBufferCatalog = RapidsBufferCatalog
       tableMeta: TableMeta,
       initialSpillPriority: Long,
       spillCallback: RapidsBuffer.SpillCallback = RapidsBuffer.defaultSpillCallback): Unit = {
-    val buffer = new RapidsDeviceMemoryBuffer(
-      id,
-      contigBuffer.getLength,
-      tableMeta,
-      Some(table),
-      contigBuffer,
-      initialSpillPriority,
-      spillCallback)
-    try {
+    closeOnExcept(
+      new RapidsDeviceMemoryBuffer(
+        id,
+        contigBuffer.getLength,
+        tableMeta,
+        Some(table),
+        contigBuffer,
+        initialSpillPriority,
+        spillCallback)) { buffer =>
       logDebug(s"Adding table for: [id=$id, size=${buffer.size}, " +
           s"meta_id=${buffer.meta.bufferMeta.id}, meta_size=${buffer.meta.bufferMeta.size}]")
       addBuffer(buffer)
-    } catch {
-      case t: Throwable =>
-        buffer.free()
```

Comment:
It needs to call `free` on the buffer, not `close`.

Reply:
Done.